xarray/core/dataset.py

```diff
@@ -24,6 +24,7 @@
 from .coordinates import (
     DatasetCoordinates, LevelCoordinatesSource,
     assert_coordinate_consistent, remap_label_indexers)
+from .duck_array_ops import datetime_to_numeric
```

Comment:
Within `differentiate`, we use `datetime_to_numeric` on Variable objects; I think you should switch to the new Variable-specific method there.